src/content/docs/en/guides/backend/turso.mdx

```diff
@@ -0,0 +1,144 @@
+- Your Database URL
+- An Access Token
+
+If you don't have an existing database and schema with Turso, create one and connect to the shell using the Turso CLI:
+
+```bash
+turso db create my-astro-db
+turso db shell my-astro-db
+```
+
+Then add the following:
+
+```sql
+CREATE TABLE posts (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    title TEXT,
+    slug TEXT UNIQUE,
+    content TEXT
+);
+```
+
+Populate the `posts` table with some initial content by **[DOING THIS THING, IDEALLY LINKING TO TURSO DOCS]**.  This is required so you can write queries and have data to fetch.
+
+### Configure environment variables
+
+Obtain your database URL using the following command:
+
+```bash
+turso db show my-astro-db --url
+```
+
+Create an auth token for the database:
+
+```bash
+turso db tokens create my-astro-db
+```
+
+Add the output from both commands above into your `.env` file at the root of your project. If this file does not exist, create one.
+
+```ini title=".env"
+TURSO_DATABASE_URL=libsql://...
+TURSO_AUTH_TOKEN=
+```
+
+:::caution
+Do not use the `PUBLIC_` prefix when creating these private [environment variables](/en/guides/environment-variables/). This will expose these values on the client.
+:::
+
+### Install LibSQL Client
+
+Install the `@libsql/client` to connect Turso to Astro:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm install @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @libsql/client
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Initialize a new client
+
+Create a file `turso.ts` in the `src` folder and invoke `createClient`, passing it `TURSO_DATABASE_URL` and `TURSO_AUTH_TOKEN`:
+
+```ts title="src/turso.ts"
+import { createClient } from "@libsql/client/web";
+
+export const turso = createClient({
+  url: import.meta.env.TURSO_DATABASE_URL,
+  authToken: import.meta.env.TURSO_AUTH_TOKEN,
+});
+```
+
+## Querying your database
+
+To access information from your database, import `turso` and [execute a SQL query](https://docs.turso.tech/libsql/client-access/javascript-typescript-sdk#execute-a-single-statement) inside any `.astro` component.
+
+The following example fetches all `posts` from your table, then displays a list of titles in a `<BlogIndex />` component:
+
+```astro title="src/components/BlogIndex.astro"
+---
+import {turso} from '../turso'
+
+const {rows} = await turso.execute('SELECT * FROM posts')
+---
+
+<ul>
+  {rows.map((post) => (
+    <li>{post.title}</li>
+  ))}
+</ul>
+```
+
+### SQL Placeholders
+
+The `execute()` method can take [an object to pass variables to the SQL statement](https://docs.turso.tech/libsql/client-access/javascript-typescript-sdk#positional-placeholders), such as `slug`, or pagination.
+
+The following example fetches a single entry from the `posts` table `WHERE` the `slug` is the retrieved value from `Astro.params`:
```

Comment:
```suggestion
The following example fetches a single entry from the `posts` table `WHERE` the `slug` is the retrieved value from `Astro.params`, then displays **....**:
```
This is great! I would say just make sure you also address what's being rendered in the template, then this works really well!